packages/cli/src/controller/migrate/manifest/migrate-manifest.controller.ts

```diff
@@ -0,0 +1,121 @@
+  }
+  return {networkFamily, chainId: getChainIdByNetworkName(networkFamily, subgraphProject.dataSources[0].network)};
+}
+
+/**
+ * Render subquery project to .ts file
+ * @param projectPath
+ * @param project
+ */
+export async function renderManifest(projectPath: string, project: CommonSubqueryProject): Promise<void> {
+  try {
+    await renderTemplate(PROJECT_TEMPLATE_PATH, projectPath, {
+      props: {
+        importTypes: {
+          network: 'ethereum',
+          projectClass: 'EthereumProject',
+          projectDatasourceKind: 'EthereumDatasourceKind',
+          projectHandlerKind: 'EthereumHandlerKind',
+        },
+        projectJson: project,
+      },
+      helper: {
+        upperFirst,
+      },
+    });
+  } catch (e) {
+    throw new Error(`When render project manifest having problems.`);
+  }
+}
+
+/**
+ *  Migrate a subgraph project manifest to subquery manifest file
+ * @param network network family
+ * @param inputPath file path to subgraph.yaml
+ * @param outputPath file path to project.ts
+ */
+export async function migrateManifest(chainInfo: ChainInfo, subgraphManifest: SubgraphProject, outputPath: string) {
+  await prepareDirPath(outputPath, false);
+  await renderManifest(outputPath, graphManifestToSubqlManifest(chainInfo, subgraphManifest));
+}
+
+/**
+ * Convert the graph project to subquery project
+ * @param network
+ * @param subgraphManifest
+ */
+function graphManifestToSubqlManifest(chainInfo: ChainInfo, subgraphManifest: SubgraphProject): CommonSubqueryProject {
+  return {
+    network: {chainId: chainInfo.chainId, endpoint: ''},
+    name: subgraphManifest.name,
+    specVersion: '1.0.0',
+    runner: {
+      node: {version: '^', name: findRunnerByNetworkFamily(chainInfo.networkFamily)},
+      query: {version: '^', name: '@subql/query'},
+    },
+    version: subgraphManifest.specVersion,
+    dataSources: subgraphDsToSubqlDs(chainInfo.networkFamily, subgraphManifest.dataSources),
+    description: subgraphManifest.description,
+    schema: subgraphManifest.schema,
+    repository: subgraphManifest.repository,
+    templates: subgraphManifest.templates
+      ? subgraphTemplateToSubqlTemplate(chainInfo.networkFamily, subgraphManifest.templates)
+      : undefined,
+  };
+}
+
+export function subgraphTemplateToSubqlTemplate(
+  network: NETWORK_FAMILY,
+  subgraphTemplates: SubgraphTemplate[]
+): TemplateKind[] {
+  const convertFunction = networkTemplateConverters[network as NETWORK_FAMILY];
+  if (!convertFunction) {
+    throw new Error(`When migration, unable find convert template method for ${network} `);
+  }
+  return subgraphTemplates.map((t) => convertFunction(t));
+}
+
+export function subgraphDsToSubqlDs(
+  network: NETWORK_FAMILY,
+  subgraphDs: SubgraphDataSource[]
+): MigrateDatasourceKind[] {
+  const convertFunction = networkDsConverters[network];
+  if (!convertFunction) {
+    throw new Error(`When migration, unable find convert dataSource method for ${network} `);
```

Comment:
Is this the first place the user would face an error if they tried an unsupported network?
```suggestion
    throw new Error(`${network} is not supported with migrations`);
```